docs/sources/reference/components/remote/remote.vault.md

```diff
@@ -40,6 +41,7 @@ Name               | Type       | Description
 `server`           | `string`   | The Vault server to connect to.                            |         | yes
 `namespace`        | `string`   | The Vault namespace to connect to (Vault Enterprise only). |         | no
 `path`             | `string`   | The path to retrieve a secret from.                        |         | yes
+`key`              | `string`   | The key to retrive a secret from.                          |         | yes
```

Comment:
I feel we need a way to allow for backwards compatibility. Likely we need to have both behaviors. If `key` is empty use existing behavior.

Nit: Typo in retrieve.

Reply:
Hi @mattdurham ,
I get that. So what I can do is wrap it in an if-else. 
And I think we have to define the argument `key` as optional. 
Thanks,
Patrick